src/Temporalio/Bridge/OptionsExtensions.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+
+namespace Temporalio.Bridge
+{
+    internal static class OptionsExtensions
+    {
+        private static readonly ByteArrayRef ClientName = ByteArrayRef.FromUTF8("temporal-dotnet");
+        private static readonly ByteArrayRef ClientVersion = ByteArrayRef.FromUTF8(
+            Temporalio.Runtime.Version
+        );
+        private static readonly ByteArrayRef SdkName = ByteArrayRef.FromUTF8("sdk-dotnet");
+
+        public static unsafe Interop.ClientOptions ToInteropOptions(
+            this Temporalio.Client.TemporalConnectionOptions options,
+            Scope scope
+        )
+        {
+            if (options.TargetHost == null)
+            {
+                throw new ArgumentException("TargetHost is required");
+            }
+            else if (options.TargetHost.Contains("://"))
+            {
+                throw new ArgumentException("TargetHost cannot have ://");
+            }
+            return new Interop.ClientOptions()
+            {
+                target_url = scope.ByteArray($"http://{options.TargetHost}"),
```

Comment:
AFAIR, when using TLS, the URL should start with `https://`.

Reply:
:+1: Will fix

Reply:
Fixed